dbms/src/Functions/FunctionsString.cpp

```diff
@@ -4188,6 +4188,54 @@ class FunctionLength : public IFunction
 private:
 };
 
+class FunctionSpace : public IFunction
+{
+public:
+    static constexpr auto name = "space";
+
+    FunctionSpace() = default;
+
+    static FunctionPtr create(const Context & /*context*/)
+    {
+        return std::make_shared<FunctionSpace>();
+    }
+
+    std::string getName() const override { return name; }
+    size_t getNumberOfArguments() const override { return 1; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (arguments.size() != 1)
+            throw Exception(
+                fmt::format("Number of arguments for function {} doesn't match: passed {}, should be 1.", getName(), arguments.size()),
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+        return std::make_shared<DataTypeString>();
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        const ColumnPtr c0_col = block.getByPosition(arguments[0]).column;
+
+        Field res_field;
+        const auto * column_space_num = checkAndGetColumn<ColumnConst>(c0_col.get());
+        Int64 space_num = 0;
+
+        if (!column_space_num)
+        {
+            space_num = column_space_num->getInt(0);
+        }
+        auto col_res = ColumnString::create();
+        col_res->reserve(space_num);
+
+        std::string res_string(space_num, ' ');
+        col_res->insert(res_string);
```

Comment:
I'm a little confused that why you only insert one row here? The row number of `col_res` should be the same as the row number of the input `block`.